### ISSUE TITLE
Casual 1 vs 1 Matchmaking Mode

### ISSUE THREAD

**Issue by Aeren1** [tetrio/issues#699]
### Checklist (issue will be disregarded & closed if incomplete!)

- [X] I've **checked both [open and recently closed issues](https://github.com/tetrio/issues/issues?q=is%3Aissue)** to ensure this feature is not already requested.
- [X] I am using the **newest version** of TETR.IO.
- [X] I am certain this is a **feature request**, not a bug.
- [X] I've **thought this feature through** properly.

-----

### Is your feature request related to a problem? Please describe.

I think Tetr.io would benefit from a casual match-made 1 vs 1 mode kind of like how League of Legends has "normal" and "ranked" modes.

### Describe the solution you'd like

What I have in mind would function similarly to Tetra League, but just no TR or visible match-making rating (there would still be an invisible rating in place to ensure fairness.) It'd serve as an environment for players to just play a few rounds without stress and to maybe try out strategies they were hesitant to in a ranked environment. 

Admittedly I don't have player stats in front of me, so I don't know if this would segment the player base too much and draw players away from the Tetra League and indirectly increase matchmaking times there.

### Describe alternatives you've considered

A work around right now is to create a custom room with a title, but this really isn't convenient. Quick Play garbage and targeting mechanics make for a really different experience.

**Comment by deedeeen** [comment]
What I have in mind is to abolish Glicko-2 from this mode and instead using the Elo rating system. You should not be affected by time and a wide variety of opponents may, and should, be encountered.

**Comment by EHwaterball** [comment]
I disagree with the component that says you shouldn't be affected by time. Time exists as a staple in many ranked systems and therefore many ranked games because during the time you don't play ranked, many variables could've changed that might affect your ranking.

From the official stats page for tetr.io:

"Some people aren't properly ranked

Some people have great days

Some people have awful days

Some people are just in the middle of massive improvement
and are on their way up to their new rank

Some people leave the game for 2 months,
"grind jstris" the entire time
and then come back to rank up

People sometimes forget that the players are humans
and not AI playing at a specific level"

Your rank is a reflection of your skill level, however when you spend time away from the game, the system has no clue where to place you at because of these things:
(a) you could have dropped in skill, 
(b) you could have practiced and had a huge increase in skill 
(c) because tetr.io ranks based on percentile, the average user in a rank could have gotten better (therefore you will lose rank)
(d) other things that affect how you play and what opponents you go against

Obviously, these three factors take a lot of time to develop. That's why it's important to play consistently and frequently: so that the system can accurately place you in a rank. If time goes by, the likelihood of these three factors increases, and the deviation part of Glicko-2 (the "time" part) is designed to take this into account. The higher this value is, the more your rank will fluctuate because the system can't accurately place you into a rank. And if you don't play for long enough: the system will have 0 clue where to put you in, losing any sense of a rank you might have (this is reflected when you lose your rank).

**Comment by deedeeen** [comment]
The idea of this issue is to hide the match ratings, therefore letting players freely play without any hassle of losing their official ranks. If you are still playing for Tetra League, you should be affected by time to correct your skill level. This is casual play, not official play. 
And just for your information, https://tetrio.team2xh.net/ is not at all _the official stats page for tetr.io_; it's only an unofficial page, fetching info once an hour. On top of it, the first sentence of the frequently asked questions said:
> This FAQ answers to questions for both TETR.IO and TETR.IO STATS, and also serves as an unofficial manual.

In short, you have severely mistaken between ranked matchmade Tetra League and unranked matchmade 1-versus-1.

**Comment by EHwaterball** [comment]
I was using "losing their official ranks" as an example as to why time is important (the system doesn't know where to place you at). For this "unranked" system there will not be a visible rank (which yeah is the point), but I do think that time is an important variable to keep. I don't see how something being unranked makes the issue of time any different.

> and a wide variety of opponents may, and should, be encountered.

I don't really get why this is important. I can see how this could be in other games, but with the linear progression of players and their skill (a person's apm, pps, and vs score is essentially what determines a placement), I can't see how the game would be anything other than one side dominating another.

Also, thanks for the correction about the stats page, but I don't think it really matters in all honesty? The part I referenced was just listing out what could go wrong with not having a time system in a ranking.

> you have severely mistaken between ranked matchmade Tetra League and unranked matchmade 1-versus-1.

That's a pretty blunt way of putting it, but I was literally only detailing why time is important in my comment (and I still don't get why it shouldn't be considered in unranked?).

**Comment by EHwaterball** [comment]
Also, you do know that time serves a different purpose other than "hiding someone's rank?" It increases how much your rank fluctuates upon a win or loss (a person with higher deviation (more time away) will get a bigger rank change). In this unranked system, yeah, hiding a rank won't exist, but I'm pretty sure that's not time's only purpose.

**Comment by hhhapz** [comment]
Personally a big fan of this. I think the non-pressure environment of a casual way to just play a 1v1 mode is very nice. I understand people can play in custom lobbies, however as mentioned previously, there is no matchmaking at play and low level players never get the opportunity to play a fair game.

**Comment by deedeeen** [comment]
I still do not quite agree - you do not have to suffer against rank inflation/rank deflation as a whole. The idea should only matter in the official TL, and this mode should be casual 1v1, not a technical challenge.

**Comment by hhhapz** [comment]
> I still do not quite agree - you do not have to suffer against rank inflation/rank deflation as a whole. The idea should only matter in the official TL, and this mode should be casual 1v1, not a technical challenge.

I assume this is a response to @EHwaterball?

**Comment by deedeeen** [comment]
That's right, @hhhapz.

**Comment by EHwaterball** [comment]
> I still do not quite agree - you do not have to suffer against rank inflation/rank deflation as a whole. The idea should only matter in the official TL, and this mode should be casual 1v1, not a technical challenge.

I don't think Glicko-2 does any of the above?

**Comment by ZXN2** [comment]
This is a cool idea, but likely can't be implemented right now. Here's why:

- "I think Tetr.io would benefit from a casual match-made 1 vs 1 mode kind of like how League of Legends has "normal" and "ranked" modes."

League of Legends can afford having two queues because they have over 4 million daily active users. TETR.IO has a peak of 3263 active users at any given time.

- "Admittedly I don't have player stats in front of me, so I don't know if this would segment the player base too much and draw players away from the Tetra League and indirectly increase matchmaking times there."

This is 100% correct. Due to the small playerbase in Tetra League, creating a normal league would drastically increase the amount of time of TL. Decay in TL is incentive to keep playing, which keeps TL slightly more active. Once the playerbase increases, a secondary queue may become an option.

I would recommend creating a custom public 1v1 lobby and changing the title to "1v1 `Z` rank," with `Z` being the desired rank you would like to 1v1.
